scripts/prepare_sector_network.py

```diff
@@ -361,71 +432,88 @@ def add_hydrogen(n, costs):
         h2_links = h2_links.groupby("buses_idx").agg(
             {"bus0": "first", "bus1": "first", "length": "mean", "capacity": "sum"}
         )
-    else:
-        attrs = ["bus0", "bus1", "length"]
-        h2_links = pd.DataFrame(columns=attrs)
 
-        candidates = pd.concat(
-            {
-                "lines": n.lines[attrs],
-                "links": n.links.loc[n.links.carrier == "DC", attrs],
-            }
-        )
-
-        for candidate in candidates.index:
-            buses = [candidates.at[candidate, "bus0"], candidates.at[candidate, "bus1"]]
-            buses.sort()
-            name = f"H2 pipeline {buses[0]} -> {buses[1]}"
-            if name not in h2_links.index:
-                h2_links.at[name, "bus0"] = buses[0]
-                h2_links.at[name, "bus1"] = buses[1]
-                h2_links.at[name, "length"] = candidates.at[candidate, "length"]
+        if len(h2_links) > 0:

```

Comment:
This if-case seems quite complex; there should be a way to simplify it.
For the first if (len > 0), it seems we can create a table like this:
| repurposing | greenfield|output|
|--|--|--|
|0|0|add_links_new_H2_pipelines|
|0|1|add_links_elec_routing_new_H2_pipelines|
|1|0|add_links_repurposed_H2_pipelines + add_links_new_H2_pipelines|
|1|1|add_links_repurposed_H2_pipelines + add_links_elec_routing_new_H2_pipelines|


Based on that, in pseudo code, we could do:
``` python
if h2_links and repurposing:
   add_links_repurposed_H2_pipelines()
if greenfield or nor h2_links:
   add_links_elec_routing_new_H2_pipelines()
else:
   add_links_new_H2_pipelines()
```
To my understanding when no h2links are available, only add_links_elec_routing_new_H2_pipelines applies.
Is that correct?

A cleaner code may also be:
``` python
if h2_links:
   if repurposing:
      add_links_repurposed_H2_pipelines()
   if greenfield:
      add_links_elec_routing_new_H2_pipelines()
   else:
      add_links_new_H2_pipelines()
else:
   add_links_new_H2_pipelines()
```

What do you think?

Reply:
Hey Davide, I think the suggestion make total sense. I implemented it already. Basically all the functionality is the same with much less cases to handle.